src/ui/MissingTerm/MissingTerms.ts

```diff
@@ -0,0 +1,223 @@
+   * This will make sure the term that are in the advance query appears in the results
+   */
+  public addTermForcedToAppear(term: string) {
+    if (this.missingTerms.indexOf(term) === -1) {
+      this.logger.warn(
+        'The term to re-inject is not present in the missing terms',
+        `You tried to inject "${term}" but the possible term to inject are: ${this.missingTerms.toString()}`,
+        'The query exited the function'
+      );
+      return;
+    }
+    this.updateTermForcedToAppear();
+    this.termForcedToAppear.push(term);
+    this.queryStateModel.set('missingTerm', [...this.termForcedToAppear]);
+  }
+
+  private updateTermForcedToAppear() {
+    this.termForcedToAppear = [...this.queryStateModel.get('missingTerm')];
+  }
+
+  private addMissingTerms() {
+    if (this.missingTerms.length === 0) {
+      return;
+    }
+    const missingTermElement = this.buildContainer();
+    if (missingTermElement) {
+      this.hideMissingTermIfNumberOfResult(missingTermElement);
+      missingTermElement.map(element => {
+        $$(this.element).append(element);
+      });
+    }
+  }
+
+  private buildContainer(): HTMLElement[] {
+    const elements: HTMLElement[] = [];
+    elements.push(this.buildCaption().el);
+    this.buildMissingTerms().forEach(term => {
+      if (term) {
+        elements.push(term.el);
+      }
+    });
+    if (elements.length > 1) {
+      return elements;
+    }
+  }
+
+  private buildCaption(): Dom {
+    return $$('span', { className: 'coveo-field-caption' }, this.options.caption);
+  }
+
+  private buildMissingTerms(): Dom[] {
+    const terms: Dom[] = this.missingTerms.map(term => {
+      if (this.hideMissingTermifFeaturedResults(term) || this.hideMissingTermIfWildcard(term)) {
+        return;
+      }
+      return this.makeTermClickableIfEnabled(term);
+    });
+    return terms;
+  }
+
+  private executeNewQuery(missingTerm: string = this.queryStateModel.get('q')) {
+    this.usageAnalytics.logSearchEvent<IAnalyticsIncludeMissingTerm>(analyticsActionCauseList.missingTermClick, {
+      missingTerm: missingTerm
+    });
+    this.queryController.executeQuery();
+  }
+
+  private makeTermClickableIfEnabled(term: string): Dom {
+    if (this.options.clickable) {
+      const termElement = $$('button', { className: 'coveo-missing-term coveo-clickable' }, term);
+      termElement.on('click', () => {
+        this.addTermForcedToAppear(term);
+        this.executeNewQuery(term);
+      });
+      return termElement;
+    } else {
+      return $$('span', { className: 'coveo-missing-term' }, term);
+    }
+  }
+
+  private createWordBoundryDelimitedRegex(term: string): RegExp {
+    return XRegExp(`${this.wordBoundary}(${term})${this.wordBoundary}`, 'g');
+  }
+
+  private hideMissingTermifFeaturedResults(term: string): boolean {
+    this.updateTermForcedToAppear();
+    return this.termForcedToAppear.indexOf(term) !== -1;
+  }
+
+  private hideMissingTermIfWildcard(term): boolean {
```

Comment:
method name is and action while it return a boolean value, something like `containsWildcard` would be more appropriate